### DISCUSSION TITLE
PWA -> SSR/PWA conversion question

### DISCUSSION THREAD

**Discussion by ssuess** [quasarframework/quasar#15213] [CLI - SSR mode] [answered]
I have a PWA that I would like to convert to SSR, mostly for the SEO benefits. There are many places in my previous code that I make use of client side / browser items like `window` and `document` (among others). I know that within the template part of my code there is the `q-no-ssr` tag I can use, but what can I do in the js part of my files? These are not things that lend themselves to being put in a boot file. Is there nothing I can use directly in the code that tells quasar this is client only?

**Comment by metalsadman** [comment, marked as answer]
you can probably check for `process.env.CLIENT` https://quasar.dev/quasar-cli-vite/handling-process-env#values-provided-by-quasar-cli

**Reply by ssuess** [reply]
Great idea, thanks!
